### PR TITLE
util/generatedexpr: migrate test-infra to testify

### LINKED ISSUE #26180: migrate test-infra to testify for `util/generatedexpr` pkg

### PR DESCRIPTION
**Pull request by nasnoisaac**
### What problem does this PR solve?

Issue Number: close #26180  

### Release note

- No release note

### CODE REVIEW

File: util/generatedexpr/gen_expr_test.go [hunk 1]
```diff
@@ -14,29 +14,30 @@
 package generatedexpr
 
 import (
-	. "github.com/pingcap/check"
+	"testing"
+
 	"github.com/pingcap/parser/ast"
+	_ "github.com/pingcap/tidb/types/parser_driver"
```
Comment (tisonkun) [inline util/generatedexpr/gen_expr_test.go]: We don't have a `TestingT(t)` so this test doesn't ever run!

**Review by tisonkun — APPROVED** [review]
LGTM.

I push one follow up commit to avoid design in advance. We do have one case for now.

**Review by xhebox — APPROVED** [review]

**Comment by tisonkun** [comment]
@wjhuang2016 my approval does not count yet since I'm not one of SIG reviewer. PTAL >_<

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**